providers/msGraph/src/lib/msgraph.provider.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  AuthenticationResult,
+  ConfidentialClientApplication,
+  Configuration,
+} from '@azure/msal-node';
+import {
+  AuthenticationProvider,
+  AuthenticationProviderOptions,
+  Client,
+} from '@microsoft/microsoft-graph-client';
+import {
+  ChannelTypeEnum,
+  IEmailOptions,
+  IEmailProvider,
+  ISendMessageSuccessResponse,
+  ICheckIntegrationResponse,
+  CheckIntegrationResponseEnum,
+} from '@novu/stateless';
+import { MSGraphConfig } from './msgraph.config';
+
+class MSGraphApiAuthenticationProvider implements AuthenticationProvider {
+  private config: Configuration;
+  constructor(conf: Configuration) {
+    this.config = conf;
+  }
+
+  public async getAccessToken(
+    options: AuthenticationProviderOptions
+  ): Promise<string> {
+    const app = new ConfidentialClientApplication(this.config);
+
+    const request = await app.acquireTokenByClientCredential({
+      scopes: ['https://graph.microsoft.com/.default'],
+    });
+
+    return request.accessToken;
+  }
+}
+
+export class MSGraphAPIProvider implements IEmailProvider {
+  id = 'msgraph';
+  channelType = ChannelTypeEnum.EMAIL as ChannelTypeEnum.EMAIL;
+  private readonly provider: MSGraphApiAuthenticationProvider;
+  private readonly client: Client;
+  private readonly clientConfig: MSGraphConfig;
+
+  constructor(private readonly setConfig: MSGraphConfig) {
```

Comment:
```suggestion
  constructor(private readonly config: MSGraphConfig) {
```
Suggestion as there is no clashing and `setConfig` gives the meaning of being a function execution an action.